src/utils/index.ts

```diff
@@ -53,18 +77,18 @@ export const is_browser = {
     Seamonkey: () => navigator.userAgent.indexOf('Seamonkey') !== -1,
 };
 
-export function stableSort(arr, compare = (a, b) => a < b) {
+export function stableSort<T>(arr: T[], compare: (a: T, b: T) => number) {
     const original = arr;
 
-    const newArray = arr.slice(0).sort((a, b) => {
+    const newArray = arr.slice(0).sort((a: T, b: T) => {
         const result = compare(a, b);
         return result === 0 ? original.indexOf(a) - original.indexOf(b) : result;
     });
 
     return newArray;
 }
 
-export function sameBar(bar1, bar2) {
+export function sameBar(bar1: any, bar2: any) {
```

Comment:
```suggestion
export function sameBar(bar1: TQuote, bar2: TQuote) {
```

please consider adding 'Volume?: number' property into the TQuote type declared in props.types.ts because a respective OHLC object in ChartIQ contains it: https://documentation.chartiq.com/tutorial-InputDataFormat.html ☺️